server/handler.go

```diff
@@ -156,7 +159,10 @@ func (h *Handler) ComQuery(
 	query string,
 	callback func(*sqltypes.Result) error,
 ) error {
-	return h.doQuery(c, query, nil, callback)
+	if err := h.doQuery(c, query, nil, callback); err != nil {
```

Comment:
Nit: just make CastSQLError handle the nil err case and you don't need the conditional logic